app/javascript/components/patient/close_contacts/CloseContactTable.js

```diff
@@ -0,0 +1,489 @@
+  };
+
+  /**
+   * Called when table is to be updated because of a query change.
+   * @param {Object} query - Updated query for table data after change.
+   */
+  handleTableUpdate = query => {
+    this.setState(
+      state => ({
+        query: { ...state.query, ...query },
+      }),
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Handles change of input in the search bar. Updates table based on input.
+   * @param {SyntheticEvent} event - Event when the search input changes
+   */
+  handleSearchChange = event => {
+    const value = event.target.value;
+    this.setState(
+      state => {
+        return { query: { ...state.query, search: value, page: 0 } };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Called when the number of entries to be shown on a page changes. Resets page to be 0.
+   * Updates state and then calls table update handler.
+   * @param {SyntheticEvent} event - Event when num entries changes
+   */
+  handleEntriesChange = event => {
+    const value = parseInt(event.target.value);
+    this.setState(
+      state => {
+        return {
+          query: { ...state.query, entries: value, page: 0 },
+        };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Called when the user clicks on the Contact Attempt for a Close Contact
+   */
+  handleContactAttempt = async data => {
+    if (await confirmDialog('Are you sure you want to log an additional contact attempt?', { title: 'New Contact Attempt' })) {
+      let currentCC = _.cloneDeep(this.state.table.rowData[`${data}`]);
+      currentCC['contact_attempts'] = currentCC['contact_attempts'] + 1;
+      this.setState(
+        {
+          showEditModal: false,
+          editRow: null,
+        },
+        () => {
+          this.updateCloseContact(currentCC, true);
+        }
+      );
+    }
+  };
+
+  /**
+   * Called when a page is clicked in the pagination component.
+   * Updates the table based on the selected page.
+   * @param {Object} page - Page object from react-paginate
+   */
+  handlePageUpdate = page => {
+    this.setState(
+      state => {
+        return {
+          query: { ...state.query, page: page.selected },
+        };
+      },
+      () => {
+        this.updateTable(this.state.query);
+      }
+    );
+  };
+
+  /**
+   * Toggle the Add New Close Contact modal by updating state.
+   */
+  toggleAddModal = () => {
+    this.setState({
+      showAddCloseContactModal: !this.state.showAddCloseContactModal,
+    });
+  };
+
+  /**
+   * Closes the Add New Close Contact modal and makes a request to add a new close contact to the db.
+   * @param {*} newCloseContactData - State from close contact modal containing needed close contact data.
+   */
+  handleAddSubmit = newCloseContactData => {
+    this.setState({ showAddCloseContactModal: false }, () => {
```

Comment:
you don't need the set state here because `updateCloseContact` reloads the page

Reply:
Disagree, im leaving this in for now.

Reply:
same with this one

Reply:
done